javascript/ql/lib/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -93,7 +93,7 @@ module UnsafeShellCommandConstruction {
       this = root.getALeaf() and
       root = isExecutedAsShellCommand(DataFlow::TypeBackTracker::end(), sys) and
       exists(string prev | prev = this.getPreviousLeaf().getStringValue() |
-        prev.regexpMatch(".* ('|\")?[0-9a-zA-Z/:_-]*")
+        prev.regexpMatch(".*\\s*('|\")?[0-9a-zA-Z/:_-]*")
```

Comment:
Can you document the intention of this?

I note that the `\\s*` term is dead* due to the greedy `.*`. Did you perhaps mean `.*\\s+`?

*: dotall makes a difference, but since there's no comment about newlines here, and the new test does not contain a newline either.

Reply:
I've realized that in generalizing the regular expression, I had created a regular expression that matched every string.  

So I've removed the regular expression, and just have a `exists(this.getPreviousLeaf().getStringValue())`, which I think is a good solution.